src/morphology.rs

```diff
@@ -349,6 +351,631 @@ pub fn close_mut(image: &mut GrayImage, norm: Norm, k: u8) {
     erode_mut(image, norm, k);
 }
 
+/// A struct representing a mask used in morphological operations
+///
+/// the mask is represented by a list of the positions of its pixels
+/// relative to its center, with a maximum distance of 255
+/// along each axis.
+/// This means that in the most extreme case, the mask could have
+/// a size of 513 by 513 pixels.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Mask {
+    elements: Vec<(i16, i16)>,
+}
+
+impl Mask {
+    /// creates a mask from a grayscale image
+    ///
+    /// a pixel is part of the mask if and only if it is non-zero
+    ///
+    /// `center_x` and `center_y` define the coordinates of the center of the mask.
+    /// They may take any value, including outside of the bounds of the input image,
+    /// but all pixels of the mask must be at most 255 pixels away from the center.
```

Comment:
What’s the reason for the restriction to a distance of 255 pixels?

Reply:
it's slightly arbitrary, but mostly to be consistent with the square(), diamond(), and disk() methods as well as the original morphology functions which all use `u8` , and because that's already way more than enough in my opinion

Reply:
I agree that 255 should be enough, but if this function would still work for arbitrary i16s I think it’s better to remove the restriction rather than require users to do any arithmetic to work out if their inputs are valid.

Reply:
that makes sense, but that's not really an option here. the user is never exposed to the `i16` actually, but what they are exposed to is the u32 type, which is the size of the image.

so there are 3 options here : 
- what i'm doing right now
- limit the size of the image without any arithmetic involved (all mask images must be 255x255 or less)
- don't limit the size of the image : then they can be as big as `u32::max`, and i need to switch the internal type from `i16` to `i64`.

currently, the center coordinates are `u8`, which means `i16`  can represent all distances as long as the image isn't too large, but no matter what type `T` i allow for the center coordinates, i will have to use a greater type internally because it needs to be able to represent `T::MAX - T::MIN` and `T::MIN - T::MAX`, and that can't be done with just `T`

Reply:
note on "limit the size of the image without any arithmetic involved" : i could put the maximum at 513x513 or even 1000x1000, as long as the distance between any pair of point along each axis is representable by an ` i16`

Reply:
A limit on the mask size without reference to center_x and center_y sounds good.

Reply:
after thinking about it, i decided to go with 513, because that's the smallest value need to be able to represent all outputs of `square()`, `disk()` and `diamond()`.

if you would prefer an other value, i don't really care too much about it.